File: typescript-connector/src/models/Transaction.ts
// Class from DFC Business Ontology: #Transaction
import { SemanticObject } from "../core/SemanticObject.js";
import { HowSubject, type HowSubjectParams } from "./HowSubject.js";
import type { Agent } from "./Agent.js";
import type { QuantitativeValue } from "./QuantitativeValue.js";

export interface TransactionParams extends HowSubjectParams {
  invoiceNumber?: string;
  quantity?: number;
  concerns?: string[];
  hasPrice?: string;
  from?: Agent | string;
  hasQuantity?: QuantitativeValue | string;
  to?: Agent | string;
}
export class Transaction extends HowSubject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Transaction";
  }

  invoiceNumber?: string;
  quantity?: number;
  concerns?: string[];
  hasPrice?: string;
  from?: Agent | string;
  hasQuantity?: QuantitativeValue | string;
  to?: Agent | string;

  constructor(
    semanticId: string,
    params?: TransactionParams,
  ) {
    super(semanticId, params);
    this.invoiceNumber = params?.invoiceNumber;
    this.quantity = params?.quantity;
    this.concerns = params?.concerns;
    this.hasPrice = params?.hasPrice;
    this.from = params?.from;
    this.hasQuantity = params?.hasQuantity;
    this.to = params?.to;
    this.semanticType = Transaction.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:invoiceNumber", () => this.invoiceNumber);
    this.registerSemanticProperty("dfc-b:quantity", () => this.quantity);
    this.registerSemanticProperty("dfc-b:concerns", () => this.concerns);
    this.registerSemanticProperty("dfc-b:hasPrice", () => this.hasPrice);
    this.registerSemanticProperty("dfc-b:from", () => this.from);
    this.registerSemanticProperty("dfc-b:hasQuantity", () => this.hasQuantity);
    this.registerSemanticProperty("dfc-b:to", () => this.to);
  }
  static {
    SemanticObject.typeRegistry.set(Transaction.SEMANTIC_TYPE, Transaction);
  }
}

File: ruby-gem/lib/models/transaction.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Transaction
require_relative '../core/semantic_object'
require_relative 'how_subject'


module DfcLinkmlConnector
  module Models
    class Transaction < HowSubject
      SEMANTIC_TYPE = "dfc-b:Transaction".freeze

      # @return [String]
      attr_accessor :invoice_number

      # @return [Float]
      attr_accessor :quantity

      # @return [String]
      attr_accessor :concerns

      # @return [String]
      attr_accessor :from

      # @return [String]
      attr_accessor :price

      # @return [String]
      attr_accessor :to

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, invoiceNumber: nil, quantity: nil, concerns: nil, from: nil, price: nil, to: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, invoiceNumber: nil, quantity: nil, concerns: nil, from: nil, price: nil, to: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @invoice_number = invoiceNumber
        @quantity = quantity
        @concerns = concerns
        @from = from
        @price = price
        @to = to
        self.semanticType = "dfc-b:Transaction"
        registerSemanticProperty("dfc-b:Transaction:invoice_number", &method("invoice_number")).valueSetter = method("invoice_number=")
        registerSemanticProperty("dfc-b:Transaction:quantity", &method("quantity")).valueSetter = method("quantity=")
        registerSemanticProperty("dfc-b:Transaction:concerns", &method("concerns")).valueSetter = method("concerns=")
        registerSemanticProperty("dfc-b:Transaction:from", &method("from")).valueSetter = method("from=")
        registerSemanticProperty("dfc-b:Transaction:has_price", &method("price")).valueSetter = method("price=")
        registerSemanticProperty("dfc-b:Transaction:to", &method("to")).valueSetter = method("to=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
